src/client/pages/Offer/utils.ts

```diff
@@ -32,3 +35,18 @@ export const getAddress = (quotes: BundledQuote[]) => {
       })
     : ''
 }
+
+export const formatRegistrationNumber = (registrationNumber: string) => {
+  if (!SE_CAR_REGISTRATION_NUMBER_REGEX.test(registrationNumber))
+    return registrationNumber
+
+  const letters = registrationNumber.match(/[A-Za-z]{3}/)?.[0] || ''
+
+  return registrationNumber.replace(letters, `${letters} `)
+}
+
+export const formatNumber = (number: number, locale: LocaleLabel) => {
+  if (typeof number !== 'number') return number
```

Comment:
Disclaimer: I'm not saying this is wrong, I'm just taking this opportunity to initiate a discussion and hopefully collect some opinions. 

This reminds a general question of mine: Isn't one of the benefits of using Typescript not having to write "type validation code" like this one?

I understand that you just want to make sure the value provided is a number since there's nothing that will help us ensure that during runtime, but in situations like that I always find myself wondering: I'm using TS and there is a "contract" about this type (GenericQuote if I'm not wrong) so if I received something different than what's defined in the "contract" wouldn't be ok "to let the app break"? 

Or maybe the idea is to find a balance... 🤔

Reply:
I totally see your point, but at the same time (as you also point out): what happens when we call this function with something that might not be a number, sent from the server? 

I guess I want to safeguard when dealing with data that we do not control, i.e from a server. I would rather have a resilient code base with some type checks here and there than to have the web application crash because a BE server sends some value somewhere which is undefined, I guess. Then on the other hand, that would lead us to have type checks everywhere, which I'm not an advocate of. I think this "slipped in" because `toLocaleString` only exists on `Number` (and `Date`). 

If this was a server, I would probably not do the type check but instead assume that a type validation on the controller would do that for me 🤔

Reply:
Yep, you've made some valid points :)